src/main/scala/is/hail/expr/Relational.scala

```diff
@@ -1639,13 +1644,31 @@ case class TableMapRows(child: TableIR, newRow: IR, newKey: Option[IndexedSeq[St
         rv2
       }
     }
-    TableValue(typ,
-      tv.globals,
-      typ.key match {
-        case Some(_) => newRVD
-        case None => newRVD.toUnpartitionedRVD
-      }
-      )
+    val newRVD = tv.rvd match {
```

Comment:
I'm somewhat confused about how to think about the divide between TableKeyBy and TableMapRows.